src/components/Modal/src/Modal.vue

```diff
@@ -86,54 +92,35 @@ const Modal = {
 
 				const deltaY = changedTouches[0].clientY - this.touchStartFrom;
 				if (deltaY > 200) {
-					this.close();
+					this.modalApi.close();
 				} else {
 					modal.style.transform = '';
 					modal.style.overflow = '';
 				}
 			}, { once: true });
 		},
 	},
-
-	render() {
-		this.assertRootComponent();
-
-		const vnode = (
-			<div
-				class={this.$s.Modal}
-				ref="modal"
-				onTouchstart={this.onTouchstart}
-			>
-				{
-					/**
-					 * @slot modal content
-					 */
-					this.$slots.default
-				}
-			</div>
-		);
-
-		// If opened with the API modal.open()
-		if (this.$vnode.context.$vnode.componentOptions.modalFunction) {
-			return vnode;
-		}
-
-		// If opened as a component (eg. Router page)
-		this.closeRoutedModal = this.modalApi.setModalVnode(vnode);
-
-		return undefined;
-	},
 };
-
-export default Modal;
-
 </script>
 
 <style module="$s">
 .Modal {
+	width: 100%;
 	height: 100%;
-	overflow-y: auto;
 	background: #f5f6f7;
-	box-shadow: 0 0 24px 8px rgba(0, 0, 0, 0.3);
+}
+
+@media screen and (min-width: 1200px) {
+	.Modal {
+		display: inline-block;
+		width: auto;
+		min-width: 600px;
+		max-width: 50vw;
```

Comment:
I'm not sure if we need both min-width: 600 and max-width: 50vw.  50vw ends up being between 600-700px so there's some scaling that happens, but around 100px. This tiny bit of scaling doesn't seem to help anything and is more likely to cause some slight hiccups with rendering content at arbitrary desktop sizes.  If we just use max-width to a set px value at desktop things are more predictable.

Reply:
Shimaa told me it should only be 600px. I've updated it.